src/NBitcoin/IBitcoinSerializable.cs

```diff
@@ -51,33 +61,48 @@ public static string ToHex(this IBitcoinSerializable serializable, Serialization
             }
         }
 
-        public static void ReadWrite(this IBitcoinSerializable serializable, byte[] bytes, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION)
-        {
-            ReadWrite(serializable, new MemoryStream(bytes), false, version);
-        }
-        public static void FromBytes(this IBitcoinSerializable serializable, byte[] bytes, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION)
-        {
-            serializable.ReadWrite(new BitcoinStream(bytes)
+		public static void ReadWrite(this IBitcoinSerializable serializable, byte[] bytes, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION, 
+            NetworkOptions options = null)
+		{
+			ReadWrite(serializable, new MemoryStream(bytes), false, version, options);
+		}
+		public static void FromBytes(this IBitcoinSerializable serializable, byte[] bytes, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION, 
+            NetworkOptions options = null)
+		{
+            var bms = new BitcoinStream(bytes)
             {
-                ProtocolVersion = version
-            });
-        }
+                ProtocolVersion = version,
+                TransactionOptions = options
+            };
+            serializable.ReadWrite(bms);
+		}
+
+		public static T Clone<T>(this T serializable, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION) where T : IBitcoinSerializable, new()
+		{
+            NetworkOptions options = NetworkOptions.All;
+			var instance = new T();
+            if (serializable is IHaveNetworkOptions)
+                options = (serializable as IHaveNetworkOptions).GetNetworkOptions();
+			instance.FromBytes(serializable.ToBytes(version, options), version, options);
+			return instance;
+		}
+        
+		public static byte[] ToBytes(this IBitcoinSerializable serializable, ProtocolVersion version = ProtocolVersion.PROTOCOL_VERSION,
+            NetworkOptions options = null)
+		{
+            // If no options have been provided then take the options from the serializable
+            if (options == null && serializable is IHaveNetworkOptions)
+                options = (serializable as IHaveNetworkOptions).GetNetworkOptions();
```

Comment:
I'm pretty sure this will give a compiler warning for unnecessary casting? Please try:

```
var haveNetworkOptions = serializable as IHaveNetworkOptions);
if (options == null && haveNetworkOptions != null)
    options = haveNetworkOptions.GetNetworkOptions();
```

Reply:
Or just specify what T should be: 

replace `where T : IBitcoinSerializable` with `where T : IHaveNetworkOptions`, inherit IHaveNetworkOptions from IBitcoinSerializable and you are good to remove this check: `if (serializable is IHaveNetworkOptions)`

Reply:
Implementing handling of a scenario that is used once or twice by changing functions that are used in  99+ cases (I'm talking about changing ToBytes and Clone) is not a good idea in my opinion.

Reply:
@quantumagi yes, @dangershony 's approach is the best actually ;) still waiting on this fix to be commited before I can approve.